### PR TITLE
Add GeoIP2 and lang middleware

### PR DESCRIPTION
**Pull request by mbeacom**
##### Description

The goal of this PR is to add geoip2 lookup based on IP address, store in the session, and track it on `UserAction`.

Added fields:

- Attr: `UserAction.ip_address` - `GenericIPAddress`
- Attr: `UserAction.geolocation_data` - `JSON`

Modified logic:

- Github middleware and view mechanisms to store geolocation and IP data to the session
- `UserAction.create` usage throughout app has implemented the new fields above.
- Added `django.middleware.locale.LocaleMiddleware` django middleware for accessing: `request.LANGUAGE_CODE`

Moving forward:

- Determine a forever home for the geoip2 logic.
- Decide on the best place to store location data longterm. (Possibly the `User` once django auth goes in).

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)

backend

##### Testing

Tested these changes locally with GH integration.

##### Refers/Fixes

Ref #651

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,15 +2,35 @@ FROM python:3.6-slim-jessie
...
+    ./bootstrap && \
+    ./configure && \
+    make && \
+    make check && \
+    make install && \
+    echo /usr/local/lib  >> /etc/ld.so.conf.d/local.conf && \
+    ldconfig
+
+# GeoIP2 Data Files
+RUN mkdir -p /usr/share/GeoIP/ && \
```
Comment (owocki) [inline Dockerfile]: hopefully these steps dont take too long and/or they're not too brittle!

File: app/github/middleware.py [hunk 2]
```diff
@@ -32,9 +39,15 @@ def process_response(self, request, response):
         token = request.session.get('access_token')
         expiration = request.session.get('access_token_last_validated')
         handle = request.session.get('handle')
+        ip_address = '24.210.224.38' if settings.DEBUG else get_real_ip(request)
```
Comment (owocki) [inline app/github/middleware.py]: note: we are behidn a load balancer and cloudfront in prod.. so we'll need to make sure `get_real_ip()` works with that

**Comment by owocki** [comment]
didnt test.. but code all looks sane to me.  hopefully theres not too much of a performance burden of the middleware

File: app/github/middleware.py (hunk 2 above)
Comment (mbeacom) [reply app/github/middleware.py]: @owocki As long as nginx is setup to forward the IP addresses appropriately, we'll be fine.  I'll take a look at the nginx config and find out.

File: Dockerfile (hunk 1 above)
Comment (mbeacom) [reply Dockerfile]: I wouldn't say they're brittle, but it does add a little extra time to the build... but they are in their own cache layer, so the user should only need to pull these files once per build environment (until they clear docker cache/system images).

**Comment by mbeacom** [comment]
@owocki If you'd like, I can modify the geoip logic to only grab the user's location on login/logout and store it as is for the duration of the session to avoid possible performance issues.
In my past experience, I never encountered any performance issues by handling geoip lookups via middleware, but we can definitely make the changes if it's a concern.

**Comment by mbeacom** [comment]
@owocki Per our discussion about performance and impact, I've removed the middleware tracking of geolocation and opted to only record location and IP when the user logs in or logs out.

Additionally, this PR adds the `Locale` django middleware that now provides us with `request.LANGUAGE_CODE` which returns the user's language code, for example: `en-us`

This should be good to merge, unless you have objections.

**Pull request merged**